### PR TITLE
[chore] Changelog and upgrade guide for v3

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

Upgrade guide and changelog for v3 release, documenting major and minor changes

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

File: UPGRADE_GUIDE.md [hunk 1]
```diff
@@ -2,6 +2,199 @@
...
+
+```go
+// Before
+batch, err := client.AddShipmentsToBatch("batch_123", shipment1.ID, shipment2.ID)
+
+// After
+batch, err := client.AddShipmentsToBatch("batch_123", shipment1, shipment2)
+```
+
+
```
Comment (Justintime50) [inline UPGRADE_GUIDE.md:197]: nit: markdown linters only want single line breaks between sections

File: UPGRADE_GUIDE.md [hunk 2]
```diff
@@ -2,6 +2,199 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-go` library between major versions.
 
+## Upgrading from 2.x to 3.0
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:5]: Resolved

File: UPGRADE_GUIDE.md [hunk 3]
```diff
@@ -2,6 +2,199 @@
...
+**Dependencies**
+
+Some dependencies had minor version bumps.
+
+## 3.0 New `DateTime` Type
+
+*Likelihood of Impact: **High***
+
+All uses of `time.Time` in the library (including in parameter and API response structs) have been replaced with a
+new `DateTime` type.
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:49]: Resolved

File: UPGRADE_GUIDE.md [hunk 4]
```diff
@@ -2,6 +2,199 @@
...
+  - `easypost.InvalidObjectError` - thrown when an object is invalid
+  - `easypost.MissingPropertyError` - thrown when a required property is missing
+  - `easypost.MissingWebhookSignatureError` - thrown when a webhook signature is missing
+  - `easypost.MismatchWebhookSignatureError` - thrown when a webhook signature does not match
+  - `easypost.ExternalApiError` - thrown when there is an error with an external API (e.g. Stripe)
+
+Any error type can be pretty-printed as a string using the `Error()` method:
+
+```go
+err := client.CreateAddress(&easypost.Address{
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:134]: Resolved

File: UPGRADE_GUIDE.md [hunk 5]
```diff
@@ -2,6 +2,199 @@
...
+- `Code` - An error code string referring to the specific error that occurred server-side
+- `StatusCode` - The HTTP status code returned by the API
+- `Message` - A human-readable message describing the error
+- `Errors` - A slice of `easypost.Error` structs, representing specific details of server-side errors encountered
+
+Any `easypost.LocalError`-derived error type will have the following properties:
+
+- `Message` - A human-readable message describing the error
+
+Common strings adn templates used to construct error messages are available as constants (e.g. `easypost.NoRatesMatchingFilters`). These can be used to perform regex-based evaluation of error messages.
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:152]: Resolved

File: UPGRADE_GUIDE.md [hunk 6]
```diff
@@ -2,6 +2,199 @@
...
+        TrackingCodes: codes,
+    })
+}
+
+// After
+code := "EZ1000000001"
+    trackers, err := client.ListTrackers(&easypost.ListTrackersOptions{
+        TrackingCode: code,
+    })
+}
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:178]: Resolved

File: UPGRADE_GUIDE.md (hunk 1 above)
Comment (nwithan8) [reply UPGRADE_GUIDE.md:197]: Resolved

File: UPGRADE_GUIDE.md [hunk 7]
```diff
@@ -2,6 +2,199 @@
...
+- `Message` - A human-readable message describing the error
+- `Errors` - A slice of `easypost.Error` structs, representing specific details of server-side errors encountered
+
+Any `easypost.LocalError`-derived error type will have the following properties:
+
+- `Message` - A human-readable message describing the error
+
+Common strings adn templates used to construct error messages are available as constants (e.g. `easypost.NoRatesMatchingFilters`). These can be used to perform regex-based evaluation of error messages.
+
+**Note**: An `easypost.LibraryError` and its subclasses represent errors that occur within this library. This is different from `easypost.Error`, which is a struct representing a server-side error returned by a failed API call.
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply UPGRADE_GUIDE.md:154]: No, per our internal discussions, this naming is valid.

- [`Error`](https://github.com/EasyPost/easypost-csharp/blob/master/EasyPost/Models/API/Error.cs) in .NET -> [`Error`](https://github.com/EasyPost/easypost-go/blob/next_major/error.go#L16) in Go
  - `Error` is not an exception, but rather an object/struct representation of an `Error` JSON schema returned by the API during a failed HTTP request.
- [`EasyPostError`](https://github.com/EasyPost/easypost-csharp/blob/master/EasyPost/Exceptions/_Base.cs#L9) in .NET -> [`LibraryError`](https://github.com/EasyPost/easypost-go/blob/next_major/error.go#L80) in Go
  - The base of all exception types in the library.
- [`ApiError`](https://github.com/EasyPost/easypost-csharp/blob/master/EasyPost/Exceptions/API/_Base.cs#L21) in .NET -> [`APIError`](https://github.com/EasyPost/easypost-go/blob/next_major/error.go#L156) in Go
  - The base of all exception types in the library related to API failures.
- No equivalent in .NET -> [`LocalError`](https://github.com/EasyPost/easypost-go/blob/next_major/error.go#L93) in Go
  - The base of all exception types in the library related to non-API failures.
  - Local, non-API errors (e.g. validation, JSON de/serialization) in .NET simply inherit `EasyPostError`. In Go, there's an intermediary `LocalError` between `LibraryError` and the specific local error class.

[the remark this answers is not in the document]
Comment (Justintime50) [reply UPGRADE_GUIDE.md:156]: This approach is wildly confusing. I wish we could follow the patterns of other libs which were much more understandable.

**Review by jchen293 — APPROVED** [review]

**Pull request merged**